File: 2018/puzzle 1.py
import re

file = open("input1.txt", "r")
frequency = 0
for line in file:
        frequency = frequency + int(line)
print "s1: " + str(frequency)

frequency = 0
map = {}
out = 0
inp = []
file = open("input1.txt", "r")
for line in file:
        inp.append(line)

while 1:        
        for line in inp:
                frequency = frequency + int(line)
                if frequency in map:
                        out = 1
                        break;
                map[frequency] = 1
        if out == 1:
                break

print "s2: " + str(frequency)

File: 2021/day1/puzzle1.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strconv"
)

func main() {
	lines := readInputLines("input.txt")
	//lines := readInputLines("../sample.txt")

	lastDepth := -1
	increases := 0

	for _, line := range lines {
		if s2i(line) > lastDepth && lastDepth >= 0 {
			increases++
		}
		lastDepth = s2i(line)
	}

	answer1 := increases
	answer2 := 0

	lastSum := -1
	for i, _ := range lines {
		if i >= len(lines) - 2 {
			break
		}
		sum := s2i(lines[i]) + s2i(lines[i+1]) + s2i(lines[i+2])
		if sum > lastSum && lastSum >= 0 {
			answer2++
		}
		lastSum = sum
	}

	fmt.Println(fmt.Sprintf("Answer 1 : %d", answer1))
	fmt.Println(fmt.Sprintf("Answer 2 : %d", answer2))
}

func s2i(val string) int {
	num, err := strconv.Atoi(val)
	if err != nil {
		fmt.Println("OH NO! OH NO! NOT AN INT!")
	}
	return num
}

func readInputLines(fileName string) []string {
	file, err := os.Open(fileName)
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	scanner := bufio.NewScanner(file)

	scanner.Split(bufio.ScanLines)
	var text []string

	for scanner.Scan() {
		text = append(text, scanner.Text())
	}

	return text
}
